wallet/cardano-wallet.js

```diff
@@ -188,37 +185,38 @@ const CardanoWallet = (secretOrMnemonic, CARDANOLITE_CONFIG) => {
   }
 
   async function getChangeAddress(usedAddressesLimit = Number.MAX_SAFE_INTEGER, offset = 0) {
-    const usedAddressesAndSecrets = await getUsedAddressesAndSecrets()
+    const usedAddressesWithHdNodes = await getUsedAddressesWithHdNodes()
 
     let result
 
-    if (usedAddressesAndSecrets.length < usedAddressesLimit) {
+    if (usedAddressesWithHdNodes.length < usedAddressesLimit) {
       const highestUsedChildIndex = Math.max(
         0x8000000,
-        ...usedAddressesAndSecrets.map((item) => item.childIndex)
+        ...usedAddressesWithHdNodes.map((item) => item.childIndex)
       )
 
-      result = (await getAddressAndSecret(highestUsedChildIndex + 1 + offset)).address
+      result = (await getAddressWithHdNode(highestUsedChildIndex + 1 + offset)).address
     } else {
       result =
-        usedAddressesAndSecrets[Math.floor(Math.random() * usedAddressesAndSecrets.length)].address
+        usedAddressesWithHdNodes[Math.floor(Math.random() * usedAddressesWithHdNodes.length)]
+          .address
     }
 
     return result
   }
 
-  async function getUnspentTxOutputsWithSecrets() {
+  async function getUnspentTxOutputsWithHdNodes() {
     let result = []
 
-    const addresses = await getUsedAddressesAndSecrets()
+    const addresses = await getUsedAddressesWithHdNodes()
 
     for (let i = 0; i < addresses.length; i++) {
       const addressUnspentOutputs = await blockchainExplorer.getUnspentTxOutputs(
         addresses[i].address
       )
 
       addressUnspentOutputs.map((element) => {
-        element.secret = addresses[i].secret
+        element.hdNode = addresses[i].hdNode
```

Comment:
uhh, this is a bit of a black magic here. First of all, I guess we should use `forEach` instead of `map` as we modify existing elements and don't really care about returned array either.
Moreover, the outer for can be similarly rewritten as `forEach`.
What do you think?

Reply:
Ok, might be more readable that way

Reply:
for ... of  suits better